server/routes/nl_interface.py

```diff
@@ -596,11 +596,14 @@ def page():
 @bp.route('/data', methods=['GET', 'POST'])
 def data():
   original_query = request.args.get('q')
-  context_history = request.get_json().get('contextHistory', [])
-  has_context = False
-  if context_history:
-    has_context = True
-  logging.info(context_history)
+  context_history = []
+  try:
+   context_history = request.get_json().get('contextHistory', [])
+   logging.info("contextHistory found:")
+   logging.info(context_history)
+  except:
+    logging.info("No contextHistory found.")
```

Comment:
we should continue executing with an empty context (assuming it's a fresh query)

Reply:
hmm. I think that is what would happen here, no? The line above the try set the contextHistory to be empty. I tested this with the direct data/ url and it produces the desired output assuming this was the first query (no context).

Reply:
Maybe check it's a POST request when get_json: `if request.method == "POST":`

Reply:
Done!